VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/Strings/StringDotFormatWithDifferentAmountOfArguments/StringDotFormatWithDifferentAmountOfArgumentsAnalyzer.cs

```diff
@@ -131,19 +181,20 @@ private void AnalyzeNode(SyntaxNodeAnalysisContext context)
             placeholderVerification:
             // Get the placeholders we use stripped off their format specifier, get the highest value 
             // and verify that this value + 1 (to account for 0-based indexing) is not greater than the amount of placeholder arguments
-            var placeholders = PlaceholderHelpers.GetPlaceholders((string) formatString.Value)
-                                                 .Cast<Match>()
-                                                 .Select(x => x.Value)
-                                                 .Select(PlaceholderHelpers.GetPlaceholderIndex)
-                                                 .Select(int.Parse)
-                                                 .ToList();
+            var placeholders = new List<int>();
+
+            foreach (Match placeholder in PlaceholderHelpers.GetPlaceholders((string) formatString.Value))
+            {
+                placeholders.Add(int.Parse(PlaceholderHelpers.GetPlaceholderIndex(placeholder.Value)));
+            }
 
             if (!placeholders.Any())
             {
                 return;
             }
 
-            var highestPlaceholder = placeholders.Max();
+            var highestPlaceholder = placeholders.Concat(new[] {int.MinValue}).Max();
```

Comment:
It should be removed since I don't see any benefit to it. It will always have a value -- there's no point in adding that min-value.